src/utils/file.ts

```diff
@@ -116,6 +116,21 @@ export async function zipDirectory(sourceDir: string, outPath: string): Promise<
     });
 }
 
+export async function zipFile(sourcePath: string, outPath: string): Promise<void> {
+    const archive = archiver("zip", { zlib: { level: 9 } });
+    const stream = fs.createWriteStream(outPath);
+
+    return new Promise((resolve, reject) => {
+        archive
+            .file(sourcePath, { name: path.basename(sourcePath) })
+            .on("error", (err) => reject(err))
+            .pipe(stream);
+
+        stream.on("close", () => resolve());
+        archive.finalize();
```

Comment:
Is there a possibility that the Promise is resolved before `archive.finalize()` is called?

I am not sure about this, maybe you can explain if I'm wrong.
```suggestion
        archive.finalize();
        stream.on("close", () => resolve()); 
```

Reply:
Even if the promise resolves, the flow will continue and `archive.finalize()` will still be called.